pam/gdm/extension_test.go

```diff
@@ -0,0 +1,248 @@
+package gdm
+
+import (
+	"slices"
+	"testing"
+	"unsafe"
+
+	"github.com/msteinert/pam"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/pam/pam_test"
+)
+
+func TestMain(t *testing.T) {
+	// We need to ensure that the the size of the data structures respects our
+	// expectations, so we do it at every
+	var msg jsonProtoMessage
+	require.Equal(t, int(unsafe.Sizeof(msg)), jsonProtoMessageSize,
+		"Unexpected request struct size, this is a fatal error")
+
+	require.Lessf(t, len(JSONProtoName), int(unsafe.Sizeof(msg.protocol_name)),
+		"protocol name '%s' exceeds the maximum size", JSONProtoName)
+}
+
+// Subtest can't be parallel as they act on global data
+//
+//nolint:tparallel
+func TestGdmExtensionSupport(t *testing.T) {
+	t.Parallel()
+	t.Cleanup(pam_test.MaybeDoLeakCheck)
+
+	testCases := map[string]struct {
+		advertisedExtensions []string
+		checkExtensions      []string
+		supportedExtensions  []string
+	}{
+		"Unknown extension is unsupported": {
+			checkExtensions:     []string{"foo.extension"},
+			supportedExtensions: nil,
+		},
+
+		"Extensions are advertised": {
+			advertisedExtensions: []string{PamExtensionCustomJSON, "foo"},
+			checkExtensions:      []string{PamExtensionCustomJSON, "foo"},
+			supportedExtensions:  []string{PamExtensionCustomJSON, "foo"},
+		},
+
+		"The private string extension unsupported if not advertised": {
+			checkExtensions:     []string{PamExtensionCustomJSON},
+			supportedExtensions: nil,
+		},
+	}
+
+	for name, tc := range testCases {
```

Comment:
Bah, I'll remove it but I honestly think that would make things quite less readable, mostly because it's a separation of the test cases data and where we actually do test things.